server/lib/src/idm/oauth2.rs

```diff
@@ -1659,14 +1678,23 @@ impl<'a> IdmServerProxyReadTransaction<'a> {
 
         // redirect_uri must be part of the client_id origin, unless the client is public and then it MAY
         // be localhost exempting it from this check and enforcement.
-        if !(o2rs.origins.contains(&auth_req.redirect_uri.origin())
+        if !((o2rs.strict_redirect_uri && o2rs.redirect_uris.contains(&auth_req.redirect_uri))
```

Comment:
This is a stylistic preference so feel free to disregard, but I'd suggest splitting this condition into several `let what_it_checks = ...` lines and then do an if over them. The reasoning is that the performance impact is minimal and it's a somewhat crucial part of code where readability would be more important than cycles saved on not short-circuiting an if condition.

Reply:
Yep, I'll update this.